inc/blocks/namespace.php

```diff
@@ -93,6 +95,47 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 		return;
 	}
 
+	update_create_xb( $xbs, $post->post_title, $post_ID, null, 'post' );
+
+}
+
+/**
+ * Synchronise any XBs on the widget page with a shadow post type.
+ *
+ * @param array $instance The array of current widget settings.
+ * @param array $new_instance The array of new widget settings.
+ * @param array $old_instance The array of old widget settings.
+ * @param WP_Widget $widget The current widget instance.
+ * @return array $instance The current widget settings.
+ */
+function on_widgets_save( $instance, $new_instance, $old_instance, $widget ) {
+
+	if ( ! isset( $instance['content'] ) ) {
+		return $instance;
+	}
+
+	// Scan for XBs in the post content.
+	$blocks = parse_blocks( $instance['content'] );
+	$xbs = find_xbs( $blocks );
+
+	// Update / Create XB's if the content contains XB's.
+	if ( ! empty( $xbs ) ) {
+		update_create_xb( $xbs, 'Widget', null, $widget->number, 'widget' );
```

Comment:
User facing text should be translatable:
```suggestion
		update_create_xb( $xbs, _( 'Widget' ), null, $widget->number, 'widget' );
```
I think core WP probably has a translation for "Widget" on its own so we can omit the text domain argument.